### PR TITLE
Add sonar-cloud support

### PR DESCRIPTION
**Pull request by aneeshafedo**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -47,3 +47,50 @@ jobs:
...
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Build with Gradle
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (NipunaMadhushan) [inline .github/workflows/pull-request.yml:83]: pass env variables as in https://github.com/ballerina-platform/openapi-tools/blob/master/.github/workflows/pull-request.yml#L25

Comment (NipunaMadhushan) [reply .github/workflows/pull-request.yml:83]: add both packageUser and packagePAT

File: build.gradle [hunk 2]
```diff
@@ -191,3 +192,11 @@ task codeCoverageReport(type: JacocoReport) {
     }
 }
 
+sonarqube {
+    properties {
+        property "sonar.projectKey", "aneeshafedo_ballerina-openapi"
+        property "sonar.organization", "aneeshafedo"
```
Comment (NipunaRanasinghe) [inline build.gradle:198]: Let's change these to the upstream repo properties, once available

Comment (aneeshafedo) [reply build.gradle]: Updated

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (aneeshafedo) [reply .github/workflows/pull-request.yml]: Updated

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**